loopchain/blockchain/block.py

```diff
@@ -295,27 +297,33 @@ def deserialize_block(self, block_dumps):
         """
         if conf.CHANNEL_OPTION[self.channel_name]["send_tx_type"] == conf.SendTxType.icx:
             dump_obj = json.loads(block_dumps)
-            self.version = dump_obj['version']
-            self.prev_block_hash = dump_obj['prev_block_hash']
-            self.merkle_tree_root_hash = dump_obj['merkle_tree_root_hash']
-            self.time_stamp = dump_obj['time_stamp']
-            self.height = dump_obj['height']
-
-            if self.height == 0:
-                validator = get_genesis_tx_validator(self.channel_name)
-                self.confirmed_transaction_list = []
-            else:
-                validator = get_tx_validator(self.channel_name)
-
-            for tx_json in dump_obj['confirmed_transaction_list']:
-                tx = validator.restore(json.dumps(tx_json))
-                self.confirmed_transaction_list.append(tx)
-
-            self.block_hash = dump_obj['block_hash']
-            self.peer_id = dump_obj['peer_id']
-            self.__signature = base64.b64decode(dump_obj['signature'].encode('UTF-8'))
-            self.__commit_state = dump_obj['commit_state'] if 'commit_state' in dump_obj else self.__commit_state
-            self.block_status = BlockStatus.confirmed
+            try:
```

Comment:
I agree, the try-catch block may be limited to `self.__next_leader_peer_id` as the except only process the `next_leader_peer_id` anyway

Reply:
I agree. I should be going to cover the problem. But I change the direction of this feature is related to that. Resolved conversation a way we couldn't be expected, because I removed the code. 😅